frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js

```diff
@@ -207,25 +217,26 @@ export class VarianceHomePage extends Component {
   handleVarianceSearch = (searchParams, clear = false) => {
     const formattedSearchParams = formatParams(searchParams);
     const persistedParams = clear ? {} : formatParams(this.state.params);
-    const updatedParams = {
-      // Start from existing state
-      ...persistedParams,
-      // Overwrite prev params with any newly provided search params
-      ...formattedSearchParams,
-      // Reset page number
-      page: String.DEFAULT_PAGE,
-      // Retain per_page if present
-      per_page: persistedParams.per_page
-        ? persistedParams.params.per_page
-        : String.DEFAULT_PER_PAGE,
-    };
-    this.setState({ params: updatedParams });
-    this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(updatedParams));
+    this.setState((prevState) => {
+      const updatedParams = {
+        // Start from existing state
+        ...persistedParams,
+        // Overwrite prev params with any newly provided search params
+        ...formattedSearchParams,
+        // Reset page number
+        page: prevState.params.page ? prevState.params.page : Strings.DEFAULT_PAGE,
+        // Retain per_page if present
+        per_page: prevState.params.per_page ? prevState.params.per_page : Strings.DEFAULT_PER_PAGE,
+      };
+      this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(updatedParams));
+      return { params: updatedParams };
+    });
   };
 
   handleVariancePageChange = (page, per_page) => {
     this.setState({ variancesLoaded: false });
     const params = { ...this.state.params, page, per_page };
+    this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(params));
     return this.props.fetchVariances(params).then(() => {
```

Comment:
`this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(params))` will cause `CompopnentWillReceiveProps' to send the new params to 'renderDataFromUrl` which will `fetchVariances`, this path will set the params correctly within the state as well as passing the correct params to the API

so, `fetchVariances` here is not needed.